test/tabs/test-firefox-tabs.js

```diff
@@ -1079,8 +1079,8 @@ exports.testNoDeadObjects = function(assert, done) {
           // 3. Arrange to close the tab once the second page loads.
           tab.on("load", function() {
             tab.close(function() {
-              var TABS = loader.sandbox("sdk/tabs/tab-firefox").TABS;
-              assert.equal(TABS.length, 0, "no dead objects");
+              var modelsFor = loader.sandbox("sdk/tabs/tab-firefox").modelsFor;
+              assert.equal(modelsFor.get(tab), undefined, "didn't retain the closed tab");
```

Comment:
this will always be true since the `tab` instance comes from a different loader instance.